lib/Transport.js

```diff
@@ -51,6 +51,7 @@ const clientVersion = require('../package.json').version
 const userAgent = `opensearch-js/${clientVersion} (${os.platform()} ${os.release()}-${os.arch()}; Node.js ${process.version})`
 const MAX_BUFFER_LENGTH = buffer.constants.MAX_LENGTH
 const MAX_STRING_LENGTH = buffer.constants.MAX_STRING_LENGTH
+const HEAP_SIZE_LIMIT = require('v8').getHeapStatistics().heap_size_limit
```

Comment:
Gotcha makes for the v8 usage then. One more clarifying question if I don't set max-old-space-size is it still static like is there a default size?

Reply:
Yes, there will be a default value ( 512 MB in latest node version i believe), and it can be different based on the system architecture. For example, the value on 64 bit system can be different from 32bit system.

Reply:
Cool. I wonder for any documentation we could provide that insight, not super explicit, or do you think that would be redundant?

Reply:
I have these noted in my design doc. Maybe I can note more down in the PR description?